### PR TITLE
dotenv setup

### PR DESCRIPTION
**Pull request by JPaul23**
**ENV setup**

- npm install dotenv.
- setup env variables and hidden to the public

### CODE REVIEW

**Comment by JPaul23** [comment]
Steps:

- pulled recent commits
- Rebased from `develop`
- Created a `config.js`
- changed the **app.js** from `const port = process.env.PORT || 3000` to `const port = config.app.port`
Now the app can run either in **dev**: development or **test**: testing environment depending to the ENV file 
below is the configurations in `config.js`
dev

File: config.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const dev = {
+  app: {
+    port: 3000
+  },
+  db: {
+    host: 'localhost',
+    port: 5432,
+    user: 'postgress',
+    name: 'phantom_dev',
+    dialect: 'postgress',
```
Comment (SimonImanigirimpuhwe) [inline config.js]: Change these values into `variables`

File: config.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+const test = {
+  app: {
+    port: 8080
+  },
+  db: {
+    host: 'localhost',
+    port: 27017,
+    user: 'postgress',
+    name: 'phantom_test',
```
Comment (SimonImanigirimpuhwe) [inline config.js]: The same here, `host`, `port`, `user` and `name` should be variables that need to be added in ```.env```

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

**Review by JPaul23 — COMMENTED** [review]
=

File: config.js (hunk 2 above)
Comment (JPaul23) [reply config.js]: Changed the structure of the config.js and created its own folder **config**. all these changes are depicted in images below.
report
report1

File: config/config.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  app: {
+    port: parseInt(process.env.APP_PORT)
+  },
+  db: {
+    host: process.env.HOST,
+    port: parseInt(process.env.DB_PORT_DEV),
+    user: process.env.DB_USER,
+    password: process.env.DB_PASSWORD,
+    name: process.env.DB_NAME_DEV
+  }
```
Comment (SimonImanigirimpuhwe) [inline config/config.js:15]: Good

**Review by SimonImanigirimpuhwe — APPROVED** [review]

**Pull request merged**